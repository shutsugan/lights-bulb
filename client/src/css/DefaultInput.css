.DefaultInput {
	border: none;
	background: transparent;
	outline: none
}