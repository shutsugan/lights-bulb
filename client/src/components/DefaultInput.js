import React, {Component} from 'react';

import '../css/DefaultInput.css';

class DefaultInput extends Component {
	constructor(props) {
		super(props);

		this.state = {
			name: this.props.name,
			id: this.props.id,
			disabled: this.props.disabled
		}
	}

	_editName = event => {
		event.target.value = event.target.value;
	}

	/**
 	 * Update the name of the current light bulb.
 	 * @param {Event} event
	 */
	_update = event => {
		const id = event.target.id;
		const data = {name: event.target.value};
		
		this.props._update(id, data);
	}

	render() {
		return(
			<input
				className="DefaultInput"
				type="text" 
				name="name"
				id={this.state.id}
				disabled={this.state.disabled}
				defaultValue={this.state.name}
				onChange={this._editName} 
				onBlur={this._update}
			/>
		);
	}
}

export default DefaultInput;