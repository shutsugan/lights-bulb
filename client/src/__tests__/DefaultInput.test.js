import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import DefaultInput from '../components/DefaultInput';

describe('DefaultInput', () => {
  it('should be defined', () => {
    expect(DefaultInput).toBeDefined();
  });

  it('should render correctly', () => {
    const tree = shallow(<DefaultInput name="test value" />);
    expect(tree).toMatchSnapshot();
  });

  it('should the value to be of type string and value equal to "test value"', () => {
    const tree = shallow(<DefaultInput name="test value" />);

    expect(typeof(tree.find('.DefaultInput').node.props.defaultValue)).toBe('string');
    expect(tree.find('.DefaultInput').node.props.defaultValue).toEqual('test value');
  });
});