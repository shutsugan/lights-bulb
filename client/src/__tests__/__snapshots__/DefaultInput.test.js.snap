// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultInput should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <input
    className="DefaultInput"
    defaultValue="test value"
    disabled={undefined}
    id={undefined}
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
/>,
  "nodes": Array [
    <input
      className="DefaultInput"
      defaultValue="test value"
      disabled={undefined}
      id={undefined}
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <DefaultInput
      name="test value"
/>,
    "_instance": DefaultInput {
      "_editName": [Function],
      "_update": [Function],
      "context": Object {},
      "props": Object {
        "name": "test value",
      },
      "refs": Object {},
      "state": Object {
        "disabled": undefined,
        "id": undefined,
        "name": "test value",
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <input
      className="DefaultInput"
      defaultValue="test value"
      disabled={undefined}
      id={undefined}
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
/>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DefaultInput
    name="test value"
/>,
}
`;
